### DISCUSSION TITLE
Package Caching

### DISCUSSION THREAD

**Discussion by keith-lackey-steamroller-tech** [AcademySoftwareFoundation/rez#1427] [Q&A]
Our main repository is on the network and am trying to get the packages to cache locally to speed things up but I can't seem to figure it out.

```
C:\Users\keith\packages.cached>rez-config cache_packages_path
C:\Users\keith\packages.cached

C:\Users\keith\packages.cached>rez-config  default_cachable
True

C:\Users\keith\packages.cached>rez-env

You are now in a rez-configured environment.

> C:\Users\keith\packages.cached>dir
 Volume in drive C is OS
 Volume Serial Number is 3456-3F03

 Directory of C:\Users\keith\packages.cached

01/04/2023  02:38 PM    <DIR>          .
01/04/2023  02:38 PM    <DIR>          ..
01/04/2023  02:38 PM    <DIR>          .sys
               0 File(s)              0 bytes
               3 Dir(s)  594,298,822,656 bytes free
```
As you can see my `cache_packages_path` variables is set to my local user directory "`packages.cached`" and my `default_cachable` is set to `True` but when i rez into my default environment it doesn't seem to trigger a cache of the packages. It has made a `.sys` folder which appears to mean the system did something but does not seem to have done anything else. What step am I overlooking?

**Comment by hughetop** [comment]
My only guess is because you went into an empty rez environment, so there was nothing to cache. Try going into a rez environment with some of your packages and see of those copy over to the cache.

**Comment by OlivierArgentieri** [comment]
I'm currently playing with rez package cache, i can detail some stuff on how i have setup things, maybe it will help.

On my rezconfig file i have add this :
```
cache_packages_path = r"<MyPathToCachedFolder>"
```
then 

## Cache per package
I have add this attrbute in one of my package, inside the package.py
```
cachable = True
```
you can see a better explanation in the documentation [here](you can see this example in the documentation [here](https://rez.readthedocs.io/en/stable/configuring_rez.html#default_cachable_per_package)
)

## default_cache per package inside rezconfig.py
If you want to centralize your default_cachable value per package you can with this:

```
default_cachable_per_package = {
    "nuke": False,
    "maya": True
}
```
you can see this example in the documentation [here](https://rez.readthedocs.io/en/stable/configuring_rez.html#default_cachable_per_package)

And this should be work
